cache4k/src/commonMain/kotlin/io/github/reactivecircus/cache4k/RealCache.kt

```diff
@@ -217,11 +219,12 @@ internal class RealCache<Key : Any, Value : Any>(
     }
 
     /**
-     * Check whether the [CacheEntry] has expired based on either access time or write time.
+     * Check whether the [CacheEntry] has expired based on either access time, write time, or set expiration time.
      */
     private fun CacheEntry<Key, Value>.isExpired(): Boolean {
-        return expiresAfterAccess && (accessTimeMark.value + expireAfterAccessDuration).hasPassedNow() ||
-            expiresAfterWrite && (writeTimeMark.value + expireAfterWriteDuration).hasPassedNow()
+        return expiresAfterAccess && (accessTimeMark.value + expireAfterAccessDuration).hasPassedNow()
+            || expiresAfterWrite && (writeTimeMark.value + expireAfterWriteDuration).hasPassedNow()
+            || expiresAt != null && expiresAt.hasPassedNow()
```

Comment:
I assume that `isExpired()` is called quite often. Maybe it is more efficient to only check `expiresAt` and update that field every time the entry is written or accessed.